mvcc: stop using 0 to specify an unbounded limit on span requests

vivekmenezes opened the pull request:
With this change all MVCC span API calls use a clear bound to limit
themselves. MVCCIterate is changed to accept a bound on
the keys it can iterate on.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/8084)

<!-- Reviewable:end -->

petermattis commented:
LGTM, though I'm not sure I agree with removing the `max == 0` special case. We use `0` in other places to indicate unlimited results. 

---

Review status: 0 of 15 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[storage/engine/mvcc.go, line 47 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNlfHgib5rWQChoHiAy:-KNlfHgjbkA-PO0O-YmX:-1351784780) ([raw file](https://github.com/cockroachdb/cockroach/blob/f0dce3504afc154e5b6fda90657e074648c67896/storage/engine/mvcc.go#L47)):*
> ```Go
> 	// MaxSpanKeys is the maximum bound on the number of keys that can be
> 	// operated on by a span (range) request.
> 	MaxSpanKeys int64 = math.MaxInt64
> ```

The term `SpanKey` is new as far as I know. Perhaps this should be named `MaxKeys`. 

---

*[storage/engine/mvcc.go, line 1497 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNlXPeTyl-U-42kYRgn:-KNlXPeTyl-U-42kYRgo:483063538) ([raw file](https://github.com/cockroachdb/cockroach/blob/f0dce3504afc154e5b6fda90657e074648c67896/storage/engine/mvcc.go#L1497)):*
> ```Go
> 	reverse bool,
> 	f func(roachpb.KeyValue) (bool, error),
> 	max int64,
> ```

When a method takes a closure (like `f`) I like that parameter to be last. For consistency with other `MVCC` methods, let's move `max` to right after `endKey`. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-:-KNlfa9Nv_V9UyBZAWqD:-486004508)*
<!-- Sent from Reviewable.io -->

tschottdorf commented on kv/split_test.go:
```diff
@@ -108,7 +108,7 @@ func TestRangeSplitMeta(t *testing.T) {
 
 	util.SucceedsSoon(t, func() error {
-		if _, _, err := engine.MVCCScan(context.Background(), s.Eng, keys.LocalMax, roachpb.KeyMax, 0, hlc.MaxTimestamp, true, nil); err != nil {
+		if _, _, err := engine.MVCCScan(context.Background(), s.Eng, keys.LocalMax, roachpb.KeyMax, engine.MaxSpanKeys, hlc.MaxTimestamp, true, nil); err != nil {
```
Do all of these packages already import `engine`? I don't think everyone should have to just to get a number.

tamird replied:
heh, how could they not? it's only used when calling into the engine, isn't it?

tschottdorf replied:
I can only see parts of this diff (epic slow internet), but I assumed the special casing would go all the way - i.e. the `BatchRequest.Header` would have to contain `math.MaxInt64` (or something like that), and I further assumed that `engine.MaxSpanKeys` would be the corresponding constant to be used everywhere. Perhaps that isn't true. I'll take a closer look when I'm back in calmer waters.

tschottdorf commented:
Ah, sorry. I overestimated the scope of this change - you're only changing the semantics of the engine-level scan operations. Then what's the upshot? I thought the discrepancy you were trying to address is that an unbounded batch doesn't come in with `math.MaxInt64` set as the limit (and I suspected you were trying to make it so). With that not being the case, I'm not sure where the added value here is.

vivekmenezes commented:
I'm trying to get rid of the 0 -> max64 -> 0 transform for the unlimited
case, and making it just 0 -> max64. I'm also trying to make users of the
API aware that they are explicitly setting an unbounded span where ever
they are using it. Using 0 is weird.

On Thu, Jul 28, 2016 at 10:05 AM Tobias Schottdorf <notifications@github.com>
wrote:

> Ah, sorry. I overestimated the scope of this change - you're only changing
> the semantics of the engine-level scan operations. Then what's the upshot?
> I thought the discrepancy you were trying to address is that an unbounded
> batch doesn't come in with math.MaxInt64 set as the limit (and I
> suspected you were trying to make it so). With that not being the case, I'm
> not sure where the added value here is.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/8084#issuecomment-235904156>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALOpBArcuPsnQDbveJS6pnoTnzUT5jThks5qaLcpgaJpZM4JW0gb>
> .
>

vivekmenezes commented:
I'm not going to continue working on this until we have enough consensus that this makes sense. No need to hurry on this one.

bdarnell commented:
Reviewed 15 of 15 files at r1.
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[storage/replica.go, line 2382 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNm4P-PS8SyUsxjgKTg:-KNm4P-PS8SyUsxjgKTh:1909691366) ([raw file](https://github.com/cockroachdb/cockroach/blob/f0dce3504afc154e5b6fda90657e074648c67896/storage/replica.go#L2382)):*
> ```Go
> 		}
> 
> 		if maxKeys != engine.MaxSpanKeys {
> ```

One of the nice things about using `MaxInt64` for "unlimited" is that you don't need `if` statements like this - just do the counting anyway; you'll never hit the limit. 

---

*[storage/replica_command.go, line 324 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNm4jzk4xqY-6rYW-bN:-KNm4jzk4xqY-6rYW-bO:1298384770) ([raw file](https://github.com/cockroachdb/cockroach/blob/f0dce3504afc154e5b6fda90657e074648c67896/storage/replica_command.go#L324)):*
> ```Go
> 	var reply roachpb.DeleteRangeResponse
> 	// TODO(vivek): replace MaxSpanKeys with a specific bound passed in.
> 	deleted, err := engine.MVCCDeleteRange(ctx, batch, ms, args.Key, args.EndKey, engine.MaxSpanKeys, h.Timestamp, h.Txn, args.ReturnKeys)
> ```

Did we not have any testing or use of `MaxEntriesToDelete`? We should remove the field from the proto at the same time we remove this use of it.

---

*[storage/replica_command.go, line 361 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNm-FxAblmDtZpeiycv:-KNm-FxAblmDtZpeiycw:1460236887) ([raw file](https://github.com/cockroachdb/cockroach/blob/a79e2bd677e746b73269ff7dac6c559e00166e2c/storage/replica_command.go#L361)):*
> ```Go
> 	args roachpb.ScanRequest,
> ) (roachpb.ScanResponse, *PostCommitTrigger, error) {
> 	if remScanResults == 0 {
> ```

@petermattis @tschottdorf This is the reason I requested this change in another PR: we're mixing zero-means-zero and zero-means-unlimited in different layers of the stack. 

I'd also be OK with -1 for unlimited, FWIW. I'm just trying to avoid dual meanings for zero. 

---

*[storage/engine/mvcc.go, line 47 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNlfHgib5rWQChoHiAy:-KNm1KnRRadyNrtXxBbg:-746166910) ([raw file](https://github.com/cockroachdb/cockroach/blob/f0dce3504afc154e5b6fda90657e074648c67896/storage/engine/mvcc.go#L47)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary>
> The term `SpanKey` is new as far as I know. Perhaps this should be named `MaxKeys`. 
</details>
`MaxSpanKeys` was a compromise suggested in the other PR because I thought `MaxRangeKeys` sounded like a per-`Range` limit and others though that `MaxKeys` sounded like it should also affect point requests somehow. 

At the `engine` level, I agree that `MaxKeys` makes the most sense, but I'm not sure we even need the constant. Why not just tell people to pass in `MaxInt64` for unlimited?  That would be pretty clear at the call sites (and works even when there is indirection between the caller and the `engine` package per @tschottdorf 's comment)

---

*[storage/engine/mvcc.go, line 1497 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNlXPeTyl-U-42kYRgn:-KNm0r02_5pxQ_TfjPty:-107242354) ([raw file](https://github.com/cockroachdb/cockroach/blob/f0dce3504afc154e5b6fda90657e074648c67896/storage/engine/mvcc.go#L1497)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary>
> When a method takes a closure (like `f`) I like that parameter to be last. For consistency with other `MVCC` methods, let's move `max` to right after `endKey`. 
</details>
I don't think `MVCCIterate` should take a `max` argument - callers that need this limit (only a couple of them do) can check it in their closures. 

---

*[storage/engine/mvcc_test.go, line 1155 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNm2hp-MKfPu0YQw5Ey:-KNm2hp-MKfPu0YQw5Ez:-1255010737) ([raw file](https://github.com/cockroachdb/cockroach/blob/f0dce3504afc154e5b6fda90657e074648c67896/storage/engine/mvcc_test.go#L1155)):*
> ```Go
> 	err = MVCCPut(context.Background(), engine, nil, testKey4, makeTS(5, 0), value1, nil)
> 
> 	kvs, _, err := MVCCScan(context.Background(), engine, testKey2, testKey4, MaxSpanKeys, makeTS(1, 0), true, nil)
> ```

If we keep the new argument, we need some tests in the engine package that use non-maxint values.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
I'll withdraw from this one. Ok with MaxInt, ok with -1. Sorry for my comments before I actually looked through the change closely.

---

Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-:-KNmBuqQ15VVwofdZ22m:-2060394176)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[storage/replica.go, line 2382 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNm4P-PS8SyUsxjgKTg:-KNmcFOkk94D4ReGw3Vc:-1995514024) ([raw file](https://github.com/cockroachdb/cockroach/blob/f0dce3504afc154e5b6fda90657e074648c67896/storage/replica.go#L2382)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> One of the nice things about using `MaxInt64` for "unlimited" is that you don't need `if` statements like this - just do the counting anyway; you'll never hit the limit. 
</details>
The downside is that whatever `executeCmd(.., maxKeys, ..)` ends up doing might be able to do stuff faster if it knows there's no limit (which it can only check by comparing against `MaxSpanKeys`).
IOW passing `MaxSpanKeys-10` might be slower than `MaxSpanKeys` depending on the operation.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084)*
<!-- Sent from Reviewable.io -->

bdarnell commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[storage/replica.go, line 2382 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNm4P-PS8SyUsxjgKTg:-KNmdEesUVR2SBWb5jv3:-1268692418) ([raw file](https://github.com/cockroachdb/cockroach/blob/f0dce3504afc154e5b6fda90657e074648c67896/storage/replica.go#L2382)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary>
> The downside is that whatever `executeCmd(.., maxKeys, ..)` ends up doing might be able to do stuff faster if it knows there's no limit (which it can only check by comparing against `MaxSpanKeys`).
> IOW passing `MaxSpanKeys-10` might be slower than `MaxSpanKeys` depending on the operation.
</details>
True. Storytime: I had a system where we tracked timeouts using doubles; I passed in DBL_MAX (~1e308) for "unlimited". This slowed everything down a great deal - it turns out that `sprintf` on DBL_MAX is extremely slow (tens of milliseconds) if you don't limit the precision, so logging became a significant portion of the request runtime. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084)*
<!-- Sent from Reviewable.io -->

vivekmenezes committed:
storage: stop using 0 to specify an unbounded limit on span requests

With this change all MVCC span API calls use a clear explictly
bound to limit themselves.

vivekmenezes commented:
Review status: 0 of 10 files reviewed at latest revision, 7 unresolved discussions, some commit checks pending.

---

*[storage/replica_command.go, line 324 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNm4jzk4xqY-6rYW-bN:-KNmYeanAPINGizcmaY_:-773384433) ([raw file](https://github.com/cockroachdb/cockroach/blob/f0dce3504afc154e5b6fda90657e074648c67896/storage/replica_command.go#L324)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> Did we not have any testing or use of `MaxEntriesToDelete`? We should remove the field from the proto at the same time we remove this use of it.
</details>
We don't and I've taken it out.

---

*[storage/engine/mvcc.go, line 47 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNlfHgib5rWQChoHiAy:-KNmq6Ck8ef4KeUNKvVY:1108472425) ([raw file](https://github.com/cockroachdb/cockroach/blob/f0dce3504afc154e5b6fda90657e074648c67896/storage/engine/mvcc.go#L47)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> `MaxSpanKeys` was a compromise suggested in the other PR because I thought `MaxRangeKeys` sounded like a per-`Range` limit and others though that `MaxKeys` sounded like it should also affect point requests somehow. 
> 
> At the `engine` level, I agree that `MaxKeys` makes the most sense, but I'm not sure we even need the constant. Why not just tell people to pass in `MaxInt64` for unlimited?  That would be pretty clear at the call sites (and works even when there is indirection between the caller and the `engine` package per @tschottdorf 's comment)
</details>
Changed to pass in MaxInt64

---

*[storage/engine/mvcc.go, line 1497 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNlXPeTyl-U-42kYRgn:-KNmmM8H0WiGtyXRpGGU:1204578049) ([raw file](https://github.com/cockroachdb/cockroach/blob/f0dce3504afc154e5b6fda90657e074648c67896/storage/engine/mvcc.go#L1497)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> I don't think `MVCCIterate` should take a `max` argument - callers that need this limit (only a couple of them do) can check it in their closures. 
</details>
Kept MVCCIterate as is and added a MVCCBoundedIterate() call that is  used by Scan/ReverseScan and will eventually by DeleteRange()

---

*[storage/engine/mvcc_test.go, line 1155 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNm2hp-MKfPu0YQw5Ey:-KNmmuNggNRICwE33XAU:366355920) ([raw file](https://github.com/cockroachdb/cockroach/blob/f0dce3504afc154e5b6fda90657e074648c67896/storage/engine/mvcc_test.go#L1155)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> If we keep the new argument, we need some tests in the engine package that use non-maxint values.
</details>
Added some tests that test a value of 0

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084)*
<!-- Sent from Reviewable.io -->

petermattis commented:
Review status: 0 of 10 files reviewed at latest revision, 8 unresolved discussions, some commit checks pending.

---

*[kv/split_test.go, line 111 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNjPvzBxPjcci6N0RvF-r1-111:-KNmyyOtcxjhVllaHukZ:891442078) ([raw file](https://github.com/cockroachdb/cockroach/blob/f0dce3504afc154e5b6fda90657e074648c67896/kv/split_test.go#L111)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> I can only see parts of this diff (epic slow internet), but I assumed the special casing would go all the way - i.e. the `BatchRequest.Header` would have to contain `math.MaxInt64` (or something like that), and I further assumed that `engine.MaxSpanKeys` would be the corresponding constant to be used everywhere. Perhaps that isn't true. I'll take a closer look when I'm back in calmer waters.
</details>
You shouldn't need a cast to `int64` here. 

---

*[storage/replica_command.go, line 361 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNm-FxAblmDtZpeiycv:-KNmzX4uA27RIFpAyZnT:1060815849) ([raw file](https://github.com/cockroachdb/cockroach/blob/a79e2bd677e746b73269ff7dac6c559e00166e2c/storage/replica_command.go#L361)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> @petermattis @tschottdorf This is the reason I requested this change in another PR: we're mixing zero-means-zero and zero-means-unlimited in different layers of the stack. 
> 
> I'd also be OK with -1 for unlimited, FWIW. I'm just trying to avoid dual meanings for zero. 
</details>
Ah, ok. In mvcc-land we're consistent that 0 means unlimited. I'm okay with `-1` or `math.MaxInt64`.

---

*[storage/engine/mvcc.go, line 47 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNlfHgib5rWQChoHiAy:-KNmzpQuoU98H-iLdEmk:723627152) ([raw file](https://github.com/cockroachdb/cockroach/blob/f0dce3504afc154e5b6fda90657e074648c67896/storage/engine/mvcc.go#L47)):*
<details><summary><i>Previously, vivekmenezes wrote…</i></summary>
> Changed to pass in MaxInt64
</details>
Looks like this is only used in tests. Perhaps remove and replace the test usage with `math.MaxInt64`.

---

*[storage/engine/mvcc.go, line 1498 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNmzwdTo--bhT2bdi0s:-KNmzwdTo--bhT2bdi0t:592805660) ([raw file](https://github.com/cockroachdb/cockroach/blob/968bcb9d67fb972ed42f732a113458d7da8e1c3e/storage/engine/mvcc.go#L1498)):*
> ```Go
> 	f func(roachpb.KeyValue) (bool, error),
> ) ([]roachpb.Intent, error) {
> 	return MVCCBoundedIterate(ctx, engine, startKey, endKey, int64(math.MaxInt64), timestamp, consistent, txn, reverse, f)
> ```

`s/int64(math.MaxInt64)/math.MaxInt64/g`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084)*
<!-- Sent from Reviewable.io -->

vivekmenezes commented:
Review status: 0 of 10 files reviewed at latest revision, 8 unresolved discussions, some commit checks pending.

---

*[kv/split_test.go, line 111 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNjPvzBxPjcci6N0RvF-r1-111:-KNnIdlRKbyTKgrb6_N1:-2034261262) ([raw file](https://github.com/cockroachdb/cockroach/blob/f0dce3504afc154e5b6fda90657e074648c67896/kv/split_test.go#L111)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary>
> You shouldn't need a cast to `int64` here. 
</details>
I didn't know you only needed a cast at var declarations:  x := int64(math.MaxInt64)

---

*[storage/engine/mvcc.go, line 47 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNlfHgib5rWQChoHiAy:-KNnIiDFF0JgZobGqWpz:677588487) ([raw file](https://github.com/cockroachdb/cockroach/blob/f0dce3504afc154e5b6fda90657e074648c67896/storage/engine/mvcc.go#L47)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary>
> Looks like this is only used in tests. Perhaps remove and replace the test usage with `math.MaxInt64`.
</details>
done

---

*[storage/engine/mvcc.go, line 1498 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNmzwdTo--bhT2bdi0s:-KNnIjoOVrhp75ckP0Cc:677588487) ([raw file](https://github.com/cockroachdb/cockroach/blob/968bcb9d67fb972ed42f732a113458d7da8e1c3e/storage/engine/mvcc.go#L1498)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary>
> `s/int64(math.MaxInt64)/math.MaxInt64/g`
</details>
done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084)*
<!-- Sent from Reviewable.io -->

vivekmenezes commented:
Any more comments on this pr? Thanks

On Thu, Jul 28, 2016, 3:35 PM Peter Mattis <notifications@github.com> wrote:

> Review status: 0 of 10 files reviewed at latest revision, 8 unresolved
> discussions, some commit checks pending.
> ------------------------------
>
> *kv/split_test.go, line 111 [r1]
> <https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNjPvzBxPjcci6N0RvF-r1-111:-KNmyyOtcxjhVllaHukZ:891442078>
> (raw file
> <https://github.com/cockroachdb/cockroach/blob/f0dce3504afc154e5b6fda90657e074648c67896/kv/split_test.go#L111>):*
> *Previously, tschottdorf (Tobias Schottdorf) wrote…*
>
> I can only see parts of this diff (epic slow internet), but I assumed the
> special casing would go all the way - i.e. the BatchRequest.Header would
> have to contain math.MaxInt64 (or something like that), and I further
> assumed that engine.MaxSpanKeys would be the corresponding constant to be
> used everywhere. Perhaps that isn't true. I'll take a closer look when I'm
> back in calmer waters.
>
>
> You shouldn't need a cast to int64 here.
>
> ------------------------------
>
> *storage/replica_command.go, line 361 [r1]
> <https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNm-FxAblmDtZpeiycv:-KNmzX4uA27RIFpAyZnT:1060815849>
> (raw file
> <https://github.com/cockroachdb/cockroach/blob/a79e2bd677e746b73269ff7dac6c559e00166e2c/storage/replica_command.go#L361>):*
> *Previously, bdarnell (Ben Darnell) wrote…*
>
> @petermattis <https://github.com/petermattis> @tschottdorf
> <https://github.com/tschottdorf> This is the reason I requested this
> change in another PR: we're mixing zero-means-zero and zero-means-unlimited
> in different layers of the stack.
>
> I'd also be OK with -1 for unlimited, FWIW. I'm just trying to avoid dual
> meanings for zero.
>
>
> Ah, ok. In mvcc-land we're consistent that 0 means unlimited. I'm okay
> with -1 or math.MaxInt64.
>
> ------------------------------
>
> *storage/engine/mvcc.go, line 47 [r1]
> <https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNlfHgib5rWQChoHiAy:-KNmzpQuoU98H-iLdEmk:723627152>
> (raw file
> <https://github.com/cockroachdb/cockroach/blob/f0dce3504afc154e5b6fda90657e074648c67896/storage/engine/mvcc.go#L47>):*
> *Previously, vivekmenezes wrote…*
>
> Changed to pass in MaxInt64
>
>
> Looks like this is only used in tests. Perhaps remove and replace the test
> usage with math.MaxInt64.
>
> ------------------------------
>
> *storage/engine/mvcc.go, line 1498 [r2]
> <https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-KNmzwdTo--bhT2bdi0s:-KNmzwdTo--bhT2bdi0t:592805660>
> (raw file
> <https://github.com/cockroachdb/cockroach/blob/968bcb9d67fb972ed42f732a113458d7da8e1c3e/storage/engine/mvcc.go#L1498>):*
>
>   f func(roachpb.KeyValue) (bool, error),
> ) ([]roachpb.Intent, error) {
>   return MVCCBoundedIterate(ctx, engine, startKey, endKey, int64(math.MaxInt64), timestamp, consistent, txn, reverse, f)
>
> s/int64(math.MaxInt64)/math.MaxInt64/g
> ------------------------------
>
> *Comments from Reviewable
> <https://reviewable.io:443/reviews/cockroachdb/cockroach/8084>*
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/8084#issuecomment-236001272>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALOpBOfYmklRCHi4aZWRsyAaDi2R1oGDks5qaQRvgaJpZM4JW0gb>
> .
>

bdarnell commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 17 files at r2, 8 of 8 files at r3.
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8084#-:-KNxvIme6twdVeBPmEhs:1417088721)*
<!-- Sent from Reviewable.io -->
